darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterStrategyFactoryImpl.java

```diff
@@ -109,18 +109,35 @@ public DarkClusterStrategy getOrCreate(@Nonnull String darkClusterName, @Nonnull
    */
   private DarkClusterStrategy createStrategy(String darkClusterName, DarkClusterConfig darkClusterConfig)
   {
-    if (darkClusterConfig.hasMultiplier() && darkClusterConfig.getMultiplier() > 0)
+    if (darkClusterConfig.hasMultiplierStrategyList())
     {
-      BaseDarkClusterDispatcher baseDarkClusterDispatcher = new BaseDarkClusterDispatcherImpl(darkClusterName, _darkClusterDispatcher,
-                                                                                              _notifier, _verifierManager);
-      return new RelativeTrafficMultiplierDarkClusterStrategy(_sourceClusterName, darkClusterName, darkClusterConfig.getMultiplier(),
-                                                              baseDarkClusterDispatcher, _notifier, _facilities.getClusterInfoProvider(), _random);
-    }
-    else
-    {
-      // Falling into this clause means that the user does not want traffic to be sent to the dark cluster, temporary or otherwise.
-      return new NoOpDarkClusterStrategy();
+      DarkClusterStrategyNameArray strategyList = darkClusterConfig.getMultiplierStrategyList();
+      for (com.linkedin.d2.DarkClusterStrategyName multiplierStrategyType : strategyList)
+      {
+        switch(multiplierStrategyType)
+        {
+          case RELATIVE_TRAFFIC:
+            if (darkClusterConfig.getMultiplier() <= 0)
+            {
+              // it is a valid case to set RELATIVE_TRAFFIC but set the multiplier to zero to
+              // temporarily set the traffic to zero; don't log anything.
+              return new NoOpDarkClusterStrategy();
+            }
```

Comment:
I'm still a fan of exposing an API in DarkClusterStrategy#isValidConfig that the factory can just call into for each strategy type :) 
That way everything is self-contained in each impl and the factory doesn't need to know these details.

Reply:
sure, I can do this. that gives us the best of both; the factory doesn't need to know these details, and we can optimize for the common codepath at the expense of just one additional call when we create the strategy (instead of at each getOrCreateStrategy)

Reply:
ended up declaring a static method in the RelativeTrafficMultiplierDarkClusterStrategy, because you can't have a static method in the interface that you can override in the impl.